editor/components/colors/with-colors.js

```diff
@@ -1,91 +1,129 @@
+			...( isString( arg ) ? { [ arg ]: kebabCase( arg ) } : arg ),
+		};
+	}, {} );
+
+	return createHigherOrderComponent(
+		compose( [
+			withSelect( ( select ) => {
 				const settings = select( 'core/editor' ).getEditorSettings();
 				return {
 					colors: get( settings, [ 'colors' ], DEFAULT_COLORS ),
 				};
 			} ),
-		( WrappedComponent ) => {
-			return class extends Component {
-				constructor() {
-					super( ...arguments );
-					/**
-					* Even though, we don't expect setAttributes or colors to change memoizing it is essential.
-					* If setAttributes or colors are not memoized, each time memoizedGetColor/memoizedSetColor are called:
-					* a new function reference is returned (even if arguments have not changed).
-					* This would make our memoized chain useless.
-					*/
-					this.memoizedGetColor = memoize( this.memoizedGetColor, { maxSize: 1 } );
-					this.memoizedSetColor = memoize( this.memoizedSetColor, { maxSize: 1 } );
-				}
+			( WrappedComponent ) => {
+				return class extends Component {
+					constructor( props ) {
+						super( props );
+
+						this.setters = this.createSetters();
 
-				memoizedGetColor( colors ) {
-					return memoize(
-						( colorName, customColorValue, colorContext ) => {
-							return {
-								name: colorName,
-								class: getColorClass( colorContext, colorName ),
-								value: getColorValue( colors, colorName, customColorValue ),
-							};
-						}
-					);
-				}
+						this.state = {};
+					}
 
-				memoizedSetColor( setAttributes, colors ) {
-					return memoize(
-						( colorNameAttribute, customColorAttribute ) => {
-							return setColorValue( colors, colorNameAttribute, customColorAttribute, setAttributes );
-						}
-					);
-				}
+					createSetters() {
+						return reduce( colorMap, ( settersAcc, colorContext, colorAttributeName ) => {
+							const ufColorAttrName = upperFirst( colorAttributeName );
+							const customColorAttributeName = `custom${ ufColorAttrName }`;
+							settersAcc[ `set${ ufColorAttrName }` ] = this.createSetColor( colorAttributeName, customColorAttributeName );
+							return settersAcc;
+						}, {} );
+					}
 
-				render() {
-					return (
-						<WrappedComponent
-							{ ...{
-								...this.props,
-								colors: undefined,
-								...mapGetSetColorToProps(
-									this.memoizedGetColor( this.props.colors ),
-									this.memoizedSetColor( this.props.setAttributes, this.props.colors ),
-									this.props
-								),
-							} }
-						/>
-					);
-				}
-			};
-		},
-	] ),
-	'withColors'
-);
+					createSetColor( colorAttributeName, customColorAttributeName ) {
+						return 	( ( colorValue ) => {
+							const colorObj = find( this.props.colors, { color: colorValue } );
+							this.props.setAttributes( {
+								[ colorAttributeName ]: colorObj && colorObj.name ? colorObj.name : undefined,
+								[ customColorAttributeName ]: colorObj && colorObj.name ? undefined : colorValue,
+							} );
+						} ).bind( this );
```

Comment:
Is `bind` valid for an arrow function? Is it needed?

Reply:
No needed, arrow functions cannot be rebinded and the bind for this is automatic. It was corrected 👍